### PR TITLE
Accept uri's with a full scheme (ssh, ftp,...)

### PR DESCRIPTION
**Pull request by adybbroe**
Currently trollmove's dispatcher expects a file-path as uri, and does not handle a full url including scheme (ssh, sftp, etc). This PR attempts to solve that.

### CODE REVIEW

File: trollmoves/dispatcher.py [hunk 1]
```diff
@@ -318,7 +321,19 @@ def run(self):
                     continue
                 destinations = self.get_destinations(msg)
                 if destinations:
-                    success = dispatch(msg.data['uri'], destinations)
+                    # Check if the url are on another host:
+                    url = urlparse(msg.data['uri'])
+                    if url.scheme != '':
+                        netloc = url.netloc
```
Comment (pnuu) [inline trollmoves/dispatcher.py:327]: This is unnecessary

File: trollmoves/dispatcher.py [hunk 2]
```diff
@@ -318,7 +321,19 @@ def run(self):
...
+                        if netloc != self.host:
+                            # This warning may appear even if the file path
+                            # does exist on both servers (a central file server
+                            # reachable from both). But in those cases one
+                            # should probably not use an url scheme but just a
+                            # file path:
+                            logger.warning(
+                                "uri is pointing to a file path on another server! Host=<%s> uri netloc=<%s>", self.host, netloc)
```
Comment (pnuu) [inline trollmoves/dispatcher.py:335]: Please shorten/wrap this line.

File: trollmoves/dispatcher.py [hunk 3]
```diff
@@ -318,7 +321,19 @@ def run(self):
                     continue
                 destinations = self.get_destinations(msg)
                 if destinations:
-                    success = dispatch(msg.data['uri'], destinations)
+                    # Check if the url are on another host:
+                    url = urlparse(msg.data['uri'])
+                    if url.scheme != '':
+                        netloc = url.netloc
+                        if netloc != self.host:
```
Comment (mraspaud) [inline trollmoves/dispatcher.py:328]: This doesn't always work. See here how it is solved:
https://github.com/pytroll/trollmoves/blob/master/trollmoves/server.py#L351-L352

Maybe a function could be made for this and used in both places ?

File: trollmoves/dispatcher.py [hunk 4]
```diff
@@ -318,7 +321,19 @@ def run(self):
...
+                            # This warning may appear even if the file path
+                            # does exist on both servers (a central file server
+                            # reachable from both). But in those cases one
+                            # should probably not use an url scheme but just a
+                            # file path:
+                            logger.warning(
```
Comment (mraspaud) [inline trollmoves/dispatcher.py:334]: Should this raise a `NotImplementedError` ?

**Comment by adybbroe** [comment]
> Couple of small comments. And please add a description :-)

Done! :-)

File: trollmoves/dispatcher.py (hunk 1 above)
Comment (adybbroe) [reply trollmoves/dispatcher.py:327]: Well, probably/strictly you are right, but I don't like this:
```
In [7]: path2 = '//san1/polar_out/direct_readout/viirs_active_fires/filtered/AFIMG_j01_d20200423_t110200.geojson'                                                      

In [8]: url2 = urlparse(path2)                                                                                                                                         

In [10]: url2.netloc                                                                                                                                                   
Out[10]: 'san1'
```
In theory `san1` could be the name of a host?

File: trollmoves/dispatcher.py (hunk 2 above)
Comment (adybbroe) [reply trollmoves/dispatcher.py:335]: Ok, will do.

File: trollmoves/dispatcher.py (hunk 4 above)
Comment (adybbroe) [reply trollmoves/dispatcher.py]: Yes, good idea!
Done

**Comment by mraspaud** [comment]
Tests are failing, I think you should use `from trollmoves.utils ...`

**Comment by adybbroe** [comment]
> Tests are failing, I think you should use `from trollmoves.utils ...`

Oh sorry!

**Comment by adybbroe** [comment]
Great thanks!

**Pull request merged**